src/Sarif.Sarifer/AnalyzeProjectCommand.cs

```diff
@@ -38,6 +38,8 @@ public AnalyzeProjectCommand(IMenuCommandService menuCommandService)
         /// </summary>
         public void Cancel()
         {
+            this.backgroundAnalysisService.ClearResultsAsync()
```

Comment:
>ClearResultsAsync [](start = 43, length = 17)

We agree that only one background analysis at a time can be in progress. But if I analyze project 1, and then I analyze project 2, don't I expect the Error List to contain the results from _both_ analyses? And if I cancel one of them, don't I expect the results from the other to remain?

This is a behavioral question so @michaelcfanning needs to decide.

Reply:
If we follow build commands, if we build the `sarif.sarifer` project, we will see 4 warnings. Then, if we build `string8`, we will see 0 warnings. So, in that case, each build clean the error list.

---
In reply to: [535426319](https://github.com/microsoft/sarif-visualstudio-extension/pull/310#discussion_r535426319) [](ancestors = 535426319)

Reply:
Ah, perfect. Thanks.

---
In reply to: [535435792](https://github.com/microsoft/sarif-visualstudio-extension/pull/310#discussion_r535435792) [](ancestors = 535435792,535426319)